PIL/GifImagePlugin.py

```diff
@@ -533,8 +533,18 @@ def getheader(im, palette=None, info=None):
 
     # Header Block
     # http://www.matthewflickinger.com/lab/whatsinagif/bits_and_bytes.asp
+
+    version = b"87a"
+    for extensionKey in ["transparency", "duration", "loop"]:
+        if info and extensionKey in info:
+            version = b"89a"
+            break
+    else:
+        if im.info.get("version") == "89a":
+            version = b"89a"
```

Comment:
We got test coverage drop report, I can suggest to move this to separate unit and add simple unit test for it. But yeah, I know that project's layout is not very friendly to create new functions. Also please let me know if you need my help.